pkg/scheduler/internal/cache/cache.go

```diff
@@ -424,13 +424,24 @@ func (cache *schedulerCache) UpdateNode(oldNode, newNode *v1.Node) error {
 
 	n, ok := cache.nodes[newNode.Name]
 	if !ok {
+		for _, cond := range newNode.Status.Conditions {
+			if cond.Type != v1.NodeReady {
+				continue
+			}
+			if cond.Status == v1.ConditionTrue {
+				break
+			}
+			klog.V(5).Infof("node %q is in NotReady status, hold off adding it to cache", newNode.Name)
```

Comment:
Do we know if `NotReady` can happen after a node is up and running, or does it only apply during startup. If it's only during startup, then is this check redundant with the one in factory.go?

Reply:
That's a very good question :) My understanding is that in theory it can still happen after a node is up and running. But in practice, I can't find/simulate a scenario to trigger that.